### PR TITLE
Only read SM Status and Activate in the dedicated functions

### PR DESCRIPTION
**Pull request by nakarlsson**
The follwing fix solved two failing CTT tests when porting SOES to the TI family. 
It also make the API cleaner since the function only does what is suppose to, not implicitly something else as well.

TF1100 Data Link Layer - Mailbox Read Serice Repeat 2
TF-1200 EtherCAT State Machine - #33  SafeOp-> Init, ErrFlag = 0, SmSett 0 or SmSett 1not match

CTT have been run on TI K2GICE and XMC43relase

### CODE REVIEW

File: soes/esc.c [hunk 1]
```diff
@@ -93,8 +93,8 @@ uint32_t ESC_ALeventread (void)
  */
```
Comment (ArthurKetels) [inline soes/esc.c]: Update comment to reflect reading of SM0ACTIVATE instead of SM0STATUS

File: soes/esc.c [hunk 2]
```diff
@@ -546,7 +536,7 @@ uint8_t ESC_mbxprocess (void)
    }
 
    /* SM0/1 access or SMn change event */
-   if (ESCvar.ALevent & ESCREG_ALEVENT_SM_MASK)
```
Comment (hefloryd) [inline soes/esc.c]: ESCREG_ALEVENT_SM_MASK is no longer used anywhere, remove?

File: README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ SOES is an EtherCAT slave stack written in c. Its purpose is to learn and
 to use. All users are invited to study the source to get an understanding
 how an EtherCAT slave functions.
 
-Features as of 2.1.2:
```
Comment (hefloryd) [inline README.md]: Could we reword this to avoid the version number? Eventually someone will forget to update it here also.

File: soes/esc.c (hunk 2 above)
Comment (ArthurKetels) [reply soes/esc.c]: Could still be used in user application. Be very careful with removing constants.

Comment (ArthurKetels) [reply soes/esc.c]: Remove "or SMn change event" in comment

Comment (hefloryd) [reply soes/esc.c]: Shouldn't the purpose of SOES be to abstract away implementation details like register contents from the user? It's no big deal though, leave it if it will be useful.

**Comment by nakarlsson** [comment]
Done
